internal/provisioner/nginx_internal.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+
+package provisioner
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mattermost/mattermost-cloud/internal/tools/aws"
+	"github.com/mattermost/mattermost-cloud/internal/tools/kops"
+	"github.com/mattermost/mattermost-cloud/model"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+type nginxInternal struct {
+	awsClient      aws.AWS
+	provisioner    *KopsProvisioner
+	kops           *kops.Cmd
+	logger         log.FieldLogger
+	cluster        *model.Cluster
+	desiredVersion string
+	actualVersion  string
+}
+
+func newNginxInternalHandle(desiredVersion string, cluster *model.Cluster, provisioner *KopsProvisioner, awsClient aws.AWS, kops *kops.Cmd, logger log.FieldLogger) (*nginxInternal, error) {
```

Comment:
Most of the utility handler logic seems like it would be identical to the external nginx utility. Is it possible to reuse most of that code or does it just make sense to copy+paste all of it? What do you think @stylianosrigas?

Reply:
@gabe I agree that we can actually reuse things like the structs and just change the helm deployment.

Reply:
After discussing this a bit with @cpanato we decided to proceed with this as is and will do some cleanup later.